mara_db/shell.py

```diff
@@ -443,17 +519,38 @@ def copy_from_stdin_command(db: object, target_table: str,
 
 @copy_from_stdin_command.register(str)
 def __(alias: str, target_table: str, csv_format: bool = None, skip_header: bool = None,
-       delimiter_char: str = None, quote_char: str = None, null_value_string: str = None, timezone: str = None):
+       delimiter_char: str = None, quote_char: str = None, null_value_string: str = None,
+       timezone: str = None, pipe_format: Format = None):
     return copy_from_stdin_command(
         dbs.db(alias), target_table=target_table, csv_format=csv_format, skip_header=skip_header,
         delimiter_char=delimiter_char, quote_char=quote_char,
-        null_value_string=null_value_string, timezone=timezone)
+        null_value_string=null_value_string, timezone=timezone, pipe_format=pipe_format)
 
 
 @copy_from_stdin_command.register(dbs.PostgreSQLDB)
 def __(db: dbs.PostgreSQLDB, target_table: str, csv_format: bool = None, skip_header: bool = None,
-       delimiter_char: str = None, quote_char: str = None, null_value_string: str = None, timezone: str = None):
-    sql = f'COPY {target_table} FROM STDIN WITH'
+       delimiter_char: str = None, quote_char: str = None, null_value_string: str = None, timezone: str = None,
+       pipe_format: Format = None):
+
+    columns = ''
+    sed_stdin = ''
+    if pipe_format:
+        if isinstance(pipe_format, JsonlFormat):
+            columns = ' (' + ', '.join(['data']) + ')'
+            # escapes JSON escapings since PostgreSQL interprets C-escapes in TEXT mode
+            sed_stdin += "sed 's/\\\\/\\\\x5C/g' \\\n| "
```

Comment:
https://xkcd.com/1638/

Maybe use `r"sed ...."` to make it not as bad as it is?